web/src/buddy_data.ts

```diff
@@ -186,6 +186,16 @@ export function user_last_seen_time_status(user_id: number): string {
     return timerender.last_seen_status_from_date(last_active_date);
 }
 
+export function fetch_presence_for_user(user_id: number): string {
```

Comment:
Also, would be good to rename this `get_user_last_seen_status` and the `fetch_user_last_seen_status` to `fetch_presence_for_user`. Also, I think instead of passing "Loading" string from this function this should just return undefined and the caller should make sure if this function returns "undefined" then "Loading..." text should be shown.